src/components/views/elements/ReplyThread.js

```diff
@@ -227,13 +228,14 @@ export default class ReplyThread extends React.Component {
 
     componentWillUnmount() {
         this.unmounted = true;
+        this.context.removeListener("Event.replaced", this.onEventReplaced);
         if (this.room) {
             this.room.removeListener("Room.redaction", this.onRoomRedaction);
             this.room.removeListener("Room.redactionCancelled", this.onRoomRedaction);
         }
     }
 
-    onRoomRedaction = (ev, room) => {
+    onEventReplaced = (ev) => {
```

Comment:
the `onRoomRedaction` and this new function are identical - might be worth shoving them through a third function which has the few lines of code in it then call that from the event handlers.